File: Java/Leaderboard/src/main/java/tddmicroexercises/leaderboard/SelfDrivingCar.java
package tddmicroexercises.leaderboard;

class SelfDrivingCar extends Driver {

    private String algorithmVersion;

    SelfDrivingCar(String algorithmVersion, String company) {
        super(algorithmVersion, company);
        this.algorithmVersion = algorithmVersion;
    }

    @Override
    String getName() {
        return String.format("Self Driving Car - %s (%s)", this.getCountry(), this.getAlgorithmVersion());
    }

    private String getAlgorithmVersion() {
        return algorithmVersion;
    }

    void setAlgorithmVersion(String algorithmVersion) {
        this.algorithmVersion = algorithmVersion;
    }

}

File: Cpp/gtest_version/leaderboard/SelfDrivingCar.cpp

#include "SelfDrivingCar.hpp"

SelfDrivingCar::SelfDrivingCar(std::string algorithmVersion, 
                               std::string company) : Driver(algorithmVersion, company), m_algorithmVersion( algorithmVersion )
{}

std::string SelfDrivingCar::getAlgorithmVersion()
{
    return m_algorithmVersion;
}

void SelfDrivingCar::setAlgorithmVersion(std::string algorithmVersion)
{
    m_algorithmVersion = algorithmVersion;
}
